README.md

```diff
@@ -401,19 +548,19 @@ just the one used on the reference implementations.
 
 ### 1.0.2. Type-Checking
 
-Type-checking is the act of preventing runtime errors by statically checking
-that certain expectations hold. For example, if a function assumes an `Int`
-argument, calling it with a `String` would be incorrect. Without types, that
-would result in a (potentially disastrous) runtime bug. Type-checking would
-prevent it by statically asserting that all functions are called with the type
-of arguments they expect.
+Type-checking is a computation step on terms that occurs in between parsing and
+evaluation which prevents runtime errors by statically checking that certain
```

Comment:
the "a computation step terms that occurs..." bit was *very* confusing even for me to understand. I'd repharse it

Reply:
Agreed, will fix